Setting up Firebase on Android (#137)
rootstrap/blog: content/Setting Up Firebase on Android- Amaury Ricardo.md

### What can we do with Firebase?

Firebase provides a lot of tools that we will talk about in next posts but resuming Firebase allow as to:
- Push notifications
- Analytics
- A/B Testing
- Authentication
- Real time db
- App distributions
- Google cloud: Hosting, Storage
- Machine Learning

And many more tools which I invite you to test.

### Initial Configuration

1. #### Adding dependencies**

On your project  Gradle file "build.gradle"

    ```java
    repositories {
        google()
        jcenter()
        maven {
            url 'https://maven.fabric.io/public'
        }
    }

    dependencies {
    		....        
    		classpath 'com.google.gms:google-services:4.3.2'  // Google Services plugin
        classpath 'io.fabric.tools:gradle:1.31.2'  // Crashlytics plugin
    		....
    }

>     ```

Reviewer: We should use gists. These don't look good in blog posts 😢

On your module (app) Gradle file "build.gradle"

    ```java
    apply plugin: 'com.android.application'

    android {
       ....
    }
    dependencies {
    		.....
    	// Add the Firebase SDK for Google Analytics
      implementation 'com.google.firebase:firebase-analytics:17.2.1'
    }

    // Add the following line to the bottom of the file
    apply plugin: 'com.google.gms.google-services'
    ```

> 2. #### reate the project on your Firebase console**

Reviewer 2: typo in `Create`. ALso you are missing the opening **

>     Open your [firebase console](https://console.firebase.google.com/)  and add a new project:

Reviewer: ```suggestion
    Open the [firebase console](https://console.firebase.google.com/) and add a new project:
```

    ![Add firebase](images/fb1.png)

    Once you create the project in Firebase console, add a new Android app:

    ![Add android plattform](images/fb2.png)

>     In the next step enter the Android applicationId , you can find it on the Gradle file:

Reviewer: ```suggestion
    In the next step, enter the Android applicationId. You can find it on the Gradle file:
```

>     In the next step enter the Android applicationId , you can find it on the Gradle file:

Reviewer: ```suggestion
    In the next step enter the Android application id. You can find it on the Gradle file:
```

    ```java
    ....
    android {
        ....
        defaultConfig {
            applicationId "com.rootstrap.android"
    		....
    ```

    **Note: In case you configure several build types with different package names you need to add an Android app to the current Firebase console for each of them.**

    ```java
    productFlavors {
            development {
                dimension "server"
                applicationIdSuffix ".dev"
            }

            production {
                dimension "server"
            }
    }
    ```

Once you create the app, download the google-services.json file and copy it to your module folder (app).

3. #### Finally run your app.

> You should be able to see on your project overview 1 daily active user (your current device or emulator)**

Reviewer 2: Extra ** at the end of the line

## That's it.

> Firebase has several tools to improve your code and productivity in order to get better performance and user experience. lay around with the console to see which ones could be helpful for your Project.

Reviewer 2: typo in `Play`.

## Setting Up Firebase on Android.
### What's Firebase?

Firebase is a mobile platform that helps you quickly develop high-quality apps, grow your user base, and more. Firebase is made up of complementary features that you can mix-and-match to fit your needs, helping you to build apps fast, without managing infrastructure. Firebase gives you functionality like analytics, databases, messaging, and crash reporting so you can move quickly and focus on your users. [[from google]](https://developer.android.com/studio/write/firebase)

> Was created by Envolved between 2011-2014, with just a couple o features (Realtime db, Hosting and Authentication) and was acquired by Google in 2014. A year later, in 2015, Google acquired Divshot, an HTML5 web-hosting platform, to merge it with Firebase, and integrated a lot of google services as well, in 2016 acquired LaunchKit a toolkit for mobile app development and in 2017 Fabric as analytics tool. Every year Google release at least one new functionality for Firebase in order to improve the mobile and web development.

Reviewer 2: Seems a bit too much and historic data I think its irrelevant for development purposes.

### What can we do with Firebase?

> Firebase provides a lot of tools that we will talk about in next posts but resuming Firebase allow as to:

Reviewer 2: `resuming` is not the word you are looking for here, maybe the copy writer can find a better headline for this sentence later.

Reviewer 3: perhaps changing the end to "but summarizing, Firebase allows as to:" (allow should have an s to show present tense)

Reviewer: `allows US to` or `allows you to`.

- Push notifications
- Analytics
- A/B Testing
- Authentication
- Real time db
- App distributions
- Google cloud: Hosting, Storage
- Machine Learning

> and a lot of others tools that I invited you to test.

Reviewer 2: I think is better: `And many more tools which I invite/recommend you to test.` to avoid repeating a lot 'a lot' 😃

> ### Setting up on Android

Reviewer 2: Initial Configuration or similar maybe? I think it would be better to avoid repeating the same title of the post.

> 1. **Adding dependencies**

Reviewer 2: Use headline tags(i.e ####) for better SEO and post structure.

On your project  Gradle file "build.gradle"

    ```java
    repositories {
        google()
        jcenter()
        maven {
            url 'https://maven.fabric.io/public'
        }
    }

    dependencies {
    		....        
    		classpath 'com.google.gms:google-services:4.3.2'  // Google Services plugin
        classpath 'io.fabric.tools:gradle:1.31.2'  // Crashlytics plugin
    		....
    }

    ```

On your module (app) Gradle file "build.gradle"

    ```java
    apply plugin: 'com.android.application'

    android {
       ....
    }
    dependencies {
    		.....
    	// Add the Firebase SDK for Google Analytics
      implementation 'com.google.firebase:firebase-analytics:17.2.1'
    }

    // Add the following line to the bottom of the file
    apply plugin: 'com.google.gms.google-services'
    ```

2. **Create the project on your Firebase console**

    Open your [firebase console](https://console.firebase.google.com/)  and add a new project:

    ![Add firebase](images/fb1.png)

>     Once you create the project in order to add a new Android app:

Reviewer 2: `once ... in order to .. doesn't make much sense. ``Once you create the project in Firebase console, add a new Android app.` or something like that.

    ![Add android plattform](images/fb2.png)

>     On the next steps enter the Android applicationId , you can found on the Gradle file:

Reviewer 2: Should be `In the next step` ..`you can find it`. Also remove space before comma.

    ```kotlin
    ....
    android {
        ....
        defaultConfig {
            applicationId "com.rootstrap.android"
    		....
    ```

    **Note: In case you configure several build types with different package names you need to add an Android app to the current Firebase console for each of them.**

    ```kotlin
    productFlavors {
            development {
                dimension "server"

>     				//Fore this case you have to create a new Android app 

Reviewer 2: typo in `For`. Also I would put this as part of the guide rather than a comment in the code.

>     				// in to your current Firabese console

Reviewer 2: Maybe just 'on the Firebase console',

    				// applicationId = com.rootstrap.android.dev
                applicationIdSuffix ".dev"
            }

            production {
                dimension "server"
            }
    }
    ```

> Once you create the app, download the google-services.json file and copy on your module folder (app).

Reviewer 2: copy or move `it to`

> 3. **Finally run your App, you should be able to see on your Project overview 1 daily active user (your current device or emulator)**

Reviewer 2: use lowercased app since it is not the actual name of the app and to keep consistency with the rest of the post.

Reviewer 2: The second part of of the sentece(after the comma) should be below the title since it is the output of the action. This way you have a shorter and clearer action 'Finally, run your app'.

## That's it.

> Firebase has several tools to improve your code and productivity in order to get better performance and user experience. Play around the console and use the ones you needed on your Project.

Reviewer 2: play around `with` the console. Maybe `Play around with the console to see which ones could be helpful for your Project.`
